backend/src/routes.js

```diff
@@ -38,41 +38,49 @@ router.get('/users/:id', async (req, res) => {
 });
 
 // endpoint: Update a user
-router.put("/users/:id", async (req, res) => {
+router.put("/users/:id", async(req, res) => {
     const user_id = req.params.id;
     const user_data = req.body;
     try {
-      await User.findByIdAndUpdate(user_id, { $set: user_data });
-      res.status(200).json({ _id: user_id });
+        await User.findByIdAndUpdate(user_id, { $set: user_data });
+        res.status(200).json({ _id: user_id });
     } catch (err) {
 
-      // change: return only the .message instead of the complete error structure
-      // res.status(404).send({ error: err });
-      res.status(404).send({ error: err.message });
+        // change: return only the .message instead of the complete error structure
+        // res.status(404).send({ error: err });
+        res.status(404).send({ error: err.message });
 
     }
 });
 
 // endpoint: insert a user
-router.post('/users', require('./middleware/middleware'), (req, res) => {
+router.post('/users', require('./middleware/middleware'), async(req, res) => {
     const newUser = new User(req.body)
     const saltRounds = 10;
-    bcrypt.hash(newUser.password, saltRounds, function(err, hash) {
-        newUser.password = hash
-        newUser.save((err, doc) => {
-            if (!err) {
-                res.status(201).json({ user: doc })
-            } else {
-                res.status(400).json({
-
-                    // change: unify returning error
-                    // message: err.message
-                    error: err.message
-
-                })
-            }
+    let newUserEmail = req.body.email
+        // console.log(newUser.email)
+    newUserEmail = await User.countDocuments({ email: newUserEmail })
+    console.log()
+    if (newUserEmail <= 0) {
+        bcrypt.hash(newUser.password, saltRounds, function(err, hash) {
+            newUser.password = hash
+            newUser.save((err, doc) => {
+                if (!err) {
+                    res.status(201).json({ user: doc })
+                } else {
+                    res.status(400).json({
+
+                        // change: unify returning error
+                        // message: err.message
+                        error: err.message
+
+                    })
+                }
+            });
         });
-    });
+    } else {
+        res.status(400).json({ error: "this email already exist" })
```

Comment:
Lets make the error message a bit prettier. Like "Could not create user. The email already exists."